lib/SILOptimizer/Transforms/DeadObjectElimination.cpp

```diff
@@ -204,6 +205,23 @@ static bool canZapInstruction(SILInstruction *Inst, bool acceptRefCountInsts) {
       isa<DeallocPartialRefInst>(Inst))
     return acceptRefCountInsts;
 
+  if (isa<InjectEnumAddrInst>(Inst))
+    return true;
+
+  // We know that the destructor has no side effects so we can remove the
+  // deallocation instruction too.
+  if (isa<DeallocationInst>(Inst) || isa<AllocationInst>(Inst))
```

Comment:
I noticed that you are checking AllocationInst here as well. That is new. Why are you changing that?

Reply:
It's needed to remove dead alloc_stack/dealloc_stack of non-trivial types.

Reply:
How did we use to handle that before? Did we just leave them in?